### PR TITLE
AN-175: Fix tests under Node package in beta-protocol-revision branch

### PR DESCRIPTION
**Pull request by acenolaza**
* This PR fixes the tests in node package after beta-protocol refactor.
* Eventhough all tests are passing there are still things to fix before considering node package to be fully up to date.
   For instance these fixes still need to be applied and tested:
   1. https://api3dao.atlassian.net/browse/AN-69
   2. https://api3dao.atlassian.net/browse/AN-98
   3. https://api3dao.atlassian.net/browse/AN-101
   4. https://api3dao.atlassian.net/browse/AN-102
   5. https://api3dao.atlassian.net/browse/AN-130
   6. others that I might have missed...
 
What changed in beta-protocol: https://docs.google.com/document/d/1dWq7e0Kle8G16wscwwwz2SgpgiuoavBvGxdpCwMMhA4/edit#heading=h.8jn8mrfz3awk

Note:
e2e and operation flow is broken because airnode is still failing to fulfill requests. I'm pretty sure is due to sending an invalid array of authorizer contract addresses and it should be fixed by AN-130. 
This is the error seen:
```
    [2021-08-30 15:58:51.266] ERROR Error: processing response error (body="{\"jsonrpc\":\"2.0\",\"id\":52,\"error\":{\"code\":-32603,\"message\":\"Error: Transaction reverted without a reason\"}}", error={"code":-32603}, requestBody="{\"method\":\"eth_call\",\"params\":[{\"to\":\"0xb7c437b267ba7ef15df34acc0d81505bf8c16056\",\"data\":\"0x8a33be0100000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000ddeeec75d3c2f2e38c541d106cbffca3f5763dd09cbf052011572b166da9cd4ff5782a51a54a7bdff5ecf71f5daf86d0890795c0eddc421714e1b46ef350e8ecf380bd0b38a40ce1a534e7ecdf4db7dbc93193530000000000000000000000005bfe3a6670f9ccaba5e89c64ddf7fff68fc9bfb10000000000000000000000006e052eede1e019da96da7b4ebd70af9999e07c1f00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000\"},\"latest\"],\"id\":52,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="http://127.0.0.1:8545/", code=SERVER_ERROR, version=web/5.1.0)
        at Logger.Object.<anonymous>.Logger.makeError (/home/sacenolaza/Projects/api3/airnode/node_modules/@ethersproject/logger/lib/index.js:180:21)
        at Logger.Object.<anonymous>.Logger.throwError (/home/sacenolaza/Projects/api3/airnode/node_modules/@ethersproject/logger/lib/index.js:189:20)
        at /home/sacenolaza/Projects/api3/airnode/node_modules/@ethersproject/web/lib/index.js:266:32
        at step (/home/sacenolaza/Projects/api3/airnode/node_modules/@ethersproject/web/lib/index.js:33:23)
        at Object.next (/home/sacenolaza/Projects/api3/airnode/node_modules/@ethersproject/web/lib/index.js:14:53)
        at fulfilled (/home/sacenolaza/Projects/api3/airnode/node_modules/@ethersproject/web/lib/index.js:5:58)
        at processTicksAndRejections (internal/process/task_queues.js:95:5)  Coordinator-ID:05f01fdb43901e40, Provider:EVM local, Chain:EVM, Chain-ID:31337
```

### CODE REVIEW

File: packages/node/src/evm/handlers/initialize-provider.test.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import * as fixtures from '../../../test/fixtures';
 
 describe('initializeProvider', () => {
   it('fetches, maps and authorizes requests', async () => {
+    jest.setTimeout(10000);
```
Comment (andreogle) [inline packages/node/src/evm/handlers/initialize-provider.test.ts:18]: Do we need this here? Things worked previously without it?

File: packages/node/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "test:e2e:watch": "SILENCE_LOGGER=true jest --coverage --watch --selectProjects e2e",
     "test": "SILENCE_LOGGER=true jest --coverage --selectProjects unit",
     "test:watch": "SILENCE_LOGGER=true jest --coverage --watch --selectProjects unit",
-    "test:watch:debug": "jest \"call-api.test\" --selectProjects unit --watch"
+    "test:watch:debug": "jest \"evm/handlers/initialize-provider.test.ts\" --selectProjects unit --watch"
```
Comment (Siegrift) [inline packages/node/package.json:21]: Nit: I think this command should not be part of package json, but a tip in README instead.

File: packages/node/test/e2e/withdrawals.feature.ts [hunk 3]
```diff
@@ -5,12 +5,12 @@ import * as fixtures from '../fixtures';
 import * as handlers from '../../src/workers/local-handlers';
 import * as wallet from '../../src/evm/wallet';
 
-it('processes withdrawals only once', async () => {
+it.only('processes withdrawals only once', async () => {
```
Comment (andreogle) [inline packages/node/test/e2e/withdrawals.feature.ts:8]: Remove the `only`
```suggestion
it('processes withdrawals only once', async () => {
```

**Review by andreogle — CHANGES_REQUESTED** [review]
Thanks for this, I can't imagine it was a fun change to implement. Just some very minor comments

**Review by Siegrift — APPROVED** [review]
:1st_place_medal: Great work. I can imagine how tiring it was to make this work.

What concerns me is that there a lot of addresses and constants needed to change. It is a bit hard to review and understand - but don't have an idea how this can be improved atm...

File: packages/node/src/evm/handlers/initialize-provider.test.ts (hunk 1 above)
Comment (acenolaza) [reply packages/node/src/evm/handlers/initialize-provider.test.ts:18]: I will revert this. When running the whole suite on my local env, this test consistently failed with a 5sec timeout but that's probably due to not having enough resources on my computer.

File: packages/node/test/e2e/withdrawals.feature.ts (hunk 3 above)
Comment (acenolaza) [reply packages/node/test/e2e/withdrawals.feature.ts:8]: Good catch!

File: packages/node/package.json (hunk 2 above)
Comment (acenolaza) [reply packages/node/package.json:21]: I agree. It is convenient but a little bit weird that one needs to change the code everytime a single test suite needs to be run. I'll remove it and add a note on the README.

**Comment by acenolaza** [comment]
>  Great work. I can imagine how tiring it was to make this work.
> 
> What concerns me is that there a lot of addresses and constants needed to change. It is a bit hard to review and understand - but don't have an idea how this can be improved atm...

Thanks Emo! What I think was most challenging was building the event logs emitted by the blockchain. Usually functions called to protocol should be idempotent and never change. I think that might be why we have so many hardcoded addresses and values. 
But there are issues like these two:
https://api3dao.atlassian.net/browse/AN-154
https://api3dao.atlassian.net/browse/AN-128
to re-organize fixtures a little bit.

**Review by andreogle — APPROVED** [review]

**Pull request merged**